### PR TITLE
[Customize Your Store] Add track events to intro page

### LINKED ISSUE #40125: Customize store intro page - Tracks
### Tracks

P2: pdnLyh-4d9-p2 (Mainly focus on high-priority tracks)

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_customize_your_store_intro_design_with_ai_click`   |
| What does the track measure      |  % of users that click on “Design with AI” |
| When does the track fire      | When a user clicks on "Design with AI” button |
| Event properties      |   |

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_customize_your_store_intro_theme_select`   |
| What does the track measure      | % of users that proceed with Tsubaki, Tazza, Zaino, or Amulet. |
| When does the track fire      | When a user selects a theme |
| Event properties     |  theme: theme slug |

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_customize_your_store_intro_browse_all_themes_click`   |
| What does the track measure      |% of users that click on “Browse all themes”. |
| When does the track fire      | When a user clicks on a “Browse all themes” button |
| Event properties     |  |

Note: After creating these tracks in code, you'll also need to PCYsg-dY7-p2 the tracks in order for them to be used in funnels and ExPlat.

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #40125.

Add tracks:

- % of users that click on “Design with AI”
- % of users that proceed with a theme
- % of users that click on “Browse all themes”.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use WP 6.3 
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper and enable `customize-store` feature flag
4. Open your browser developer console and run `localStorage.setItem( 'debug', 'wc-admin:*' );` to set debug for tracks
5. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store`
6. Click on `Design with AI`
7. Observe that `wcadmin_customize_your_store_intro_design_with_ai_click` event is recorded
8. Go back to intro page
9. Click on TT1 theme
10. Observe that `wcadmin_wcadmin_customize_your_store_intro_theme_select ` event event is recorded with theme = `twentytwentyone` and is_active = `yes`
11. Reload the page
12. Click on other theme
13. Observe that `wcadmin_wcadmin_customize_your_store_intro_theme_select ` event event is recorded with theme = `[theme you selected]` and is_active = `no`
14. Reload the page
15. Click on `Browse all themes` button
16. Observe that `wcadmin_customize_your_store_intro_browse_all_themes_click` event is recorded 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/intro/actions.ts [hunk 1]
```diff
@@ -23,3 +25,24 @@ export const assignThemeCards = assign<
...
 	},
 } );
+
+export const recordTracksDesignWithAIClicked = () => {
+	recordEvent( 'customize_your_store_intro_design_with_ai_click' );
+};
+
+export const recordTracksThemeSelected = (
+	_context: customizeStoreStateMachineContext,
+	event: Extract<
```
Comment (rjchow) [inline plugins/woocommerce-admin/client/customize-store/intro/actions.ts:35]: ooo nice

File: plugins/woocommerce-admin/client/customize-store/intro/services.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export const fetchThemeCards = async () => {
 			description: 'The default theme for WordPress.',
 			image: 'https://i0.wp.com/s2.wp.com/wp-content/themes/pub/twentytwentyone/screenshot.png',
 			styleVariations: [],
+			isActive: true,
```
Comment (rjchow) [inline plugins/woocommerce-admin/client/customize-store/intro/services.ts:11]: this feels like a footgun that could come back to bite us, should we leave isActive out until we put in the logic to determine what theme is active?

**Review by rjchow — APPROVED** [review]
Nice work!

File: plugins/woocommerce-admin/client/customize-store/intro/services.ts (hunk 2 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/customize-store/intro/services.ts:11]: We will replace this entirely with an API call to fetch theme data, so I think it should be fine. 🙂

**Review by adrianduffell — APPROVED** [review]
It tested perfectly, thanks @chihsuan 🚀

**Pull request merged**